### DISCUSSION TITLE
Digging - Base Class

### DISCUSSION THREAD

**Discussion by collinsadrian** [codecadwallader/codemaid#772] [Ideas]
Hi, and first of all what a great tool you have created. Just found it and love what you have done.

I was just wondering whether, while digging a file, you could also include the base class in to the view. I have had a look through all the settings and can't seem to find anything in relation to this.

I understand that the reordering etc. wouldn't work, but being able to view a class with all the base class members would really help (me).

Thanks

**Comment by codecadwallader** [comment]
Thanks for the kind comments, I appreciate it. :)

There is not currently a way to enable viewing the base class in the Spade tool window.  Usually we rely on a file to be open as that provides more information and API access.  A few things that would need to be figured out:

1.  How to identify the base class (is it available through the FileCodeModel of the child?)
2.  Does the base class need to be an open document in the editor to get access to the same level of information?
3.  What would the UI look like to distinguish the base class from the child class?

There is a similar enhancement request in the backlog #601

**Reply by collinsadrian** [reply]
Have given this some thought...

1. The Base Class, and perhaps the interfaces, would appear at the top of the spade window in there own collapsed menu.
2. Just the name of the base class/interface would be listed with an appropriate icon prefixed and a plus icon suffixed.
3. Clicking on the name of the file would open that file, but not give it focus.
4. The plus sign would be removed to indicate it has been opened and included.
5. As you now have the file open, you could interrogate the file an include fields, members etc. in with the content of the parent class and suffix [base] 
6. If it was an interface opened you could annotate which current members/fields etc. belong to the interface by prefixing [interface name]

I have probably trivialised a lot here, so apologies if I don't make much sense or do not understand the complexities involved.

Thanks

**Reply by collinsadrian** [reply]
May be a setting to open base class and/or interface files automatically :)
